src/algebra/lie/basic.lean

```diff
@@ -281,36 +281,36 @@ def trans (e₁ : L₁ ≃ₗ⁅R⁆ L₂) (e₂ : L₂ ≃ₗ⁅R⁆ L₃) : L
 
 end equiv
 
-namespace direct_sum
-open dfinsupp
-open_locale direct_sum
+end lie_algebra
 
-variables {R : Type u} [comm_ring R]
-variables {ι : Type v} {L : ι → Type w}
-variables [Π i, lie_ring (L i)] [Π i, lie_algebra R (L i)]
-
-/-- The direct sum of Lie rings carries a natural Lie ring structure. -/
-instance : lie_ring (⨁ i, L i) :=
-{ bracket     := zip_with (λ i, λ x y, ⁅x, y⁆) (λ i, lie_zero 0),
-  add_lie     := λ x y z, by { ext, simp only [zip_with_apply, add_apply, add_lie], },
-  lie_add     := λ x y z, by { ext, simp only [zip_with_apply, add_apply, lie_add], },
-  lie_self    := λ x, by { ext, simp only [zip_with_apply, add_apply, lie_self, zero_apply], },
-  leibniz_lie := λ x y z, by { ext, simp only [direct_sum.sub_apply,
-    zip_with_apply, add_apply, zero_apply], apply leibniz_lie, },
-  ..(infer_instance : add_comm_group _) }
-
-@[simp] lemma bracket_apply {x y : (⨁ i, L i)} {i : ι} :
-  ⁅x, y⁆ i = ⁅x i, y i⁆ := zip_with_apply _ _ x y i
-
-/-- The direct sum of Lie algebras carries a natural Lie algebra structure. -/
-instance : lie_algebra R (⨁ i, L i) :=
-{ lie_smul := λ c x y, by { ext, simp only [
-    zip_with_apply, direct_sum.smul_apply, bracket_apply, lie_smul] },
-  ..(infer_instance : module R _) }
-
-end direct_sum
+section lie_module_morphisms
 
-end lie_algebra
+variables (R : Type u) (L : Type v) {M : Type w} {N : Type w₁}
+variables [comm_ring R] [lie_ring L] [lie_algebra R L]
+variables [add_comm_group M] [add_comm_group N] [module R M] [module R N]
+variables [lie_ring_module L M] [lie_ring_module L N] [lie_module R L M] [lie_module R L N]
+
+/-- The condition for a linear map between two modules to be a morphism of Lie modules. -/
+class is_lie_module_hom (f : M →ₗ[R] N) : Prop :=
```

Comment:
What is the reason for making this a type class predicate on bundled linear maps? Why not a bundled structure extending linear maps?

Reply:
I have no good reason for not bundling beyond the fact that I have recently grown a little suspicious that we may bundle too much (forcing us to write lots of boiler plate for coercions, `norm_cast`, `simp` lemmas etc).

I guess I was privately thinking it might be useful to see how things would develop with the unbundled approach but I will happily switch these to be bundled if you prefer. It would certainly better parallel the objects which are all bundled.

Reply:
If you bundle, then `map_lie` can become a simp-lemma. If you don't bundle, then it can't.
The boilerplate is a serious problem. We should look into options for auto-generating it. But it seems that all people with automation skills are currently very busy.

Reply:
OK I'll swap to the bundled approach, hopefully this evening.

Reply:
Now done.